src/lang/modifyAst.test.ts

```diff
@@ -638,11 +639,27 @@ describe('Testing removeSingleConstraintInfo', () => {
         code.indexOf(lineOfInterest) + lineOfInterest.length,
       ]
       const pathToNode = getNodePathFromSourceRange(ast, range)
+      let argPosition: SimplifiedArgDetails
+      if (key === 'arrayIndex' && typeof value === 'number') {
+        argPosition = {
+          type: 'arrayItem',
+          index: value === 0 ? 0 : 1,
+        }
+      } else if (key === 'objectProperty' && typeof value === 'string') {
+        argPosition = {
+          type: 'objectProperty',
+          key: value as InputArgKeys,
+        }
```

Comment:
If we used type narrowing here would it be necessary to check the `value` type? Shouldn't it already be the correct type?

Reply:
This is just the type inference coming from the vite utils is not fantastic, I can massage this.

Reply:
ah simple fix `as const` 8e5a99acb387afc20ad235d1ff61e72864040fc8